### PR TITLE
Add value_stats logging option to show statistical information about …

### PR DESCRIPTION
**Pull request by RyanUnderhill**
Adds a new logging option "value_stats" that is on by default. This will show 'value_stats' after any array of values is also logged.

So when tensor inputs/output data is shown, the 'value_stats' will be shown after them. Also for things like 'model_logits' as below:

```
  model_logits   Values[ 2.51172 4.83594 6.32422 1.30469 1.97949 ... -1.28418 -1.28418 -1.28711 -1.28516 -1.28223 ]
  value_stats    Min: -5.35156 Max: 8.35156 Mean: 0.933448 StdDev: 1.1297
```

image

This will also detect non-finite values and say the index of the first value similar to this image: (was renamed to just 'value_stats' but the rest is accurate)

image

This also generalizes the Float16ToFloat32 code to be templatized on the number of bits per field. This makes the implementation easier to read (no more magic numbers) and also adds bfloat16 support trivially.

### CODE REVIEW

File: src/models/debugging.cpp [hunk 1]
```diff
@@ -61,6 +113,30 @@ void DumpValues(std::ostream& stream, ONNXTensorElementDataType type, const void
...
+  if (g_log.value_stats) {
+    Stats stats;
+    if (type == Ort::TypeToTensorType<float>) {
+      auto p_values = reinterpret_cast<const float*>(p_values_raw);
+      for (size_t i = 0; i < count; i++)
+        stats << p_values[i];
+    } else if (type == Ort::TypeToTensorType<double>) {
+      auto p_values = reinterpret_cast<const double*>(p_values_raw);
+      for (size_t i = 0; i < count; i++)
+        stats << static_cast<float>(p_values[i]);
```
Comment (aciddelgado) [inline src/models/debugging.cpp:126]: possible to introduce inf values when making this conversion?

Comment (RyanUnderhill) [reply src/models/debugging.cpp:126]: Yep, and then we'll report that we did in the stat class. We convert to float for our scoring, so this is good.

Comment (RyanUnderhill) [reply src/models/debugging.cpp:126]: (also, I don't think any models use doubles for logits)

**Review by aciddelgado — APPROVED** [review]
